crates/core/src/protocols/evalcheck/prove.rs

```diff
@@ -585,6 +571,66 @@ where
 					transcript,
 				)?;
 			}
+			MultilinearPolyVariant::Squared(id) => {
+				let position = if let Some(position) = self
+					.square_roots_memoization
+					.iter()
+					.position(|(ep, _, _)| *ep == eval_point)
+				{
+					transcript.message().write(&(position as u32));
+					position
+				} else {
+					let position = self.square_roots_memoization.len();
+					transcript.message().write(&(position as u32));
+
+					let sqrt_eval_point = eval_point
```

Comment:
pub fn sqrt_eval_point<F: BinaryField, P: PackedField<Scalar = F>>(eval_point: &[F]) -> Vec<F> {
	let n_vars = eval_point.len();
	let mut acc = pack_slice::<P>(eval_point);
	for acc in &mut acc {
		for _ in 0..(P::Scalar::DEGREE - 1) {
			*acc = acc.square();
		}
	}
	P::iter_slice(&acc).take(n_vars).collect()
}